tools/eslint-rules/no-useless-regex-char-class-escape.js

```diff
@@ -0,0 +1,192 @@
+/**
+ * @fileoverview Disallow useless escape in regex character class
+ * Based on 'no-useless-escape' rule
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const REGEX_CHARCLASS_ESCAPES = new Set('\\bcdDfnrsStvwWxu0123456789]');
+
+/**
+ * Parses a regular expression into a list of regex character class list.
+ * @param {string} regExpText raw text used to create the regular expression
+ * @returns {Object[]} A list of character classes tokens with index and
+ *   escape info
+ * @example
+ *
+ * parseRegExpCharClass('a\\b[cd-]')
+ *
+ * returns:
+ * [
+ *   {
+ *     empty: false,
+ *     start: 4,
+ *     end: 6,
+ *     chars: [
+ *       {text: 'c', index: 4, escaped: false},
+ *       {text: 'd', index: 5, escaped: false},
+ *       {text: '-', index: 6, escaped: false}
+ *     ]
+ *   }
+ * ]
+ */
+
+function parseRegExpCharClass(regExpText) {
```

Comment:
I noticed that this function is very different from the [corresponding one in `no-useless-escape`](https://github.com/eslint/eslint/blob/dbff37a680217a0034518398cf467540b651e735/lib/rules/no-useless-escape.js#L47-L70). This isn't necessarily a problem, but is there a reason to change it?